gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -464,3 +469,179 @@ def _get_variant_calls(self, record, formats):
         call.info[field] = data
       calls.append(call)
     return calls
+
+
+class NucleusParser(VcfParser):
+  """An Iterator for processing a single VCF file using Nucleus."""
+
+  def __init__(self,
+               file_name,  # type: str
+               range_tracker,  # type: range_trackers.OffsetRangeTracker
+               file_pattern,  # type: str
+               compression_type,  # type: str
+               allow_malformed_records,  # type: bool
+               representative_header_lines=None,  # type:  List[str]
+               **kwargs  # type: **str
+              ):
+    # type: (...) -> None
+    super(NucleusParser, self).__init__(file_name,
+                                        range_tracker,
+                                        file_pattern,
+                                        compression_type,
+                                        allow_malformed_records,
+                                        representative_header_lines,
+                                        **kwargs)
+    # This member will be properly initiated in _init_with_header().
+    self._vcf_reader = None
+    # These members will be properly initiated in _extract_header_fields().
+    self._is_info_repeated = {}
```

Comment:
Please create a dict of info and format that point to `VcfInfo` and `VcfFormatInfo` instead; that is more future-proof (e.g. if we want to also check the description or type for another purpose, we could just reuse those dicts later on without introducing more specialized dicts). You can keep the is_repeated arg in `_convert_list_value` as it's a nice abstraction, but you can make a helper function that given a number string would return true/false if the field is repeated.